client/src/common/types/entities.ts

```diff
@@ -2,20 +2,33 @@
 
 export type Category = number | string | boolean;
 
+export type ColumnSchema = AnnotationColumn | LayoutColumn | DataFrame;
+
 export interface AnnotationColumn {
   categories?: Category[];
   name: string;
   type: "string" | "float32" | "int32" | "categorical" | "boolean";
   writable: boolean;
 }
 
-interface DataFrame {
+export interface DataFrame {
```

Comment:
See above comment re: `Schema`.  This is not a DataFrame, but a schema sub-element describing the X matrix schema.

Reply:
Updated in #2375.